gradle/build-cache-configuration.settings.gradle.kts

```diff
@@ -0,0 +1,78 @@
+import java.net.URI
+
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * This script is applied to the settings in buildSrc and the main build. That is why we
+ * need this to be a script unless we can model dual usage better with composite/included builds or another solution.
+ */
+
+/* Currently IntelliJ assumes that script plugins configure project objects. That is why we need to declare the receiver
+ * object for the functions explicitly to have IDE support. The build would run successfully without declaring
+ * the receiver object, if this script is applied from a settings.gradle.kts
+ *
+ * Note that this script has the ending .settings.gradle.kts. This will eventually used to make IntelliJ aware that
+ * this is configuring a settings object.
+ *
+ */
+
+
+
+val remoteCacheUrl = System.getProperty("gradle.cache.remote.url")?.let { URI(it) }
```

Comment:
Should we check for the empty String here before constructing the URL? That's at least what the old code did.

Reply:
I'm not sure. It is our build, so we have not made a contract with the world. I would not let the Groovy truth definition necessarily make all our Kotlin scripts more complex just because we want to replicate exactly the same behavior.

Reply:
Your call.

Reply:
We consider empty project properties in many places as being set, i.e. they pass the being set validation. So this would be no different here. It would be even good to have this consistency.

Reply:
@wolfs @lptr Please approve this pull request or let me know that you want the script to check for empty string. The Rodrigo can review whether this is idiomatic Kotlin and we can merge.

Reply:
Empty property should be fine.